docs/core/sdk/file-based-apps.md

```diff
@@ -0,0 +1,406 @@
+    "http": {
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
+      "applicationUrl": "http://localhost:5000",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "https": {
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
+      "applicationUrl": "https://localhost:5001;http://localhost:5000",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    }
+  }
+}
+```
+
+### Multiple file-based apps
+
+When you have multiple file-based apps in the same directory, each app can have its own launch settings file:
+
+```Directory
+📁 myapps/
+├── foo.cs
+├── foo.run.json
+├── bar.cs
+└── bar.run.json
+```
+
+### Profile selection
+
+The .NET CLI selects launch profiles using the following priority:
+
+1. The profile specified by the `--launch-profile` option.
+1. The profile specified by the `DOTNET_LAUNCH_PROFILE` environment variable.
+1. The first profile defined in the launch settings file.
+
+To run with a specific profile:
+
+```dotnetcli
+dotnet run app.cs --launch-profile https
+```
+
+### Traditional launch settings
+
+File-based apps also support the traditional `Properties/launchSettings.json` file. If both files exist, the traditional location takes priority. If both files are present, the .NET CLI logs a warning to clarify which file is used.
+
+## Shell execution
+
+Enable direct execution of file-based apps on Unix-like systems by using a shebang line and executable permissions.
+
+Add a shebang at the top of your file:
+
+```csharp
+#!/usr/bin/env dotnet
+#:package Spectre.Console
+
+using Spectre.Console;
+
+AnsiConsole.MarkupLine("[green]Hello, World![/]");
+```
+
+Make the file executable:
+
+```bash
+chmod +x file.cs
+```
+
+Run directly:
+
+```bash
+./file.cs
+```
+
+## Implicit build files
+
+File-based apps respect MSBuild and NuGet configuration files in the same directory or parent directories. These files affect how the SDK builds your application. Be mindful of these files when organizing your file-based apps.
+
+### `Directory.Build.props`
+
+Defines MSBuild properties that apply to all projects in a directory tree. File-based apps inherit these properties.
+
+### `Directory.Build.targets`
+
+Defines MSBuild targets and custom build logic. File-based apps execute these targets during build.
+
+### `Directory.Packages.props`
+
+Enables central package management for NuGet dependencies. File-based apps can use centrally managed package versions.
+
+### `nuget.config`
+
+Configures NuGet package sources and settings. File-based apps use these configurations when restoring packages.
+
+### `global.json`
+
+Specifies the .NET SDK version to use. File-based apps respect this version selection.
+
+## Build caching
+
+The .NET SDK caches build outputs to improve performance on subsequent builds. File-based apps participate in this caching system.
```

Comment:
This feature is unique to file-based apps, so it's less that "file-based apps participate in this caching system", and more that the caching system is **unique** to file-based apps.